Rules/UseVerboseMessageInDSCResource.cs

```diff
@@ -22,70 +22,58 @@
 namespace Microsoft.Windows.PowerShell.ScriptAnalyzer.BuiltinRules
 {
     /// <summary>
-    /// ProvideVerboseMessage: Analyzes the ast to check that Write-Verbose is called at least once in every cmdlet or script.
+    /// ProvideVerboseMessage: Analyzes the ast to check that Write-Verbose is called for DSC Resources
     /// </summary>
-    [Export(typeof(IScriptRule))]
-    public class ProvideVerboseMessage : SkipNamedBlock, IScriptRule
+    [Export(typeof(IDSCResourceRule))]
+    public class UseVerboseMessageInDSCResource : SkipNamedBlock, IDSCResourceRule
     {
         /// <summary>
-        /// AnalyzeScript: Analyzes the ast to check that Write-Verbose is called at least once in every cmdlet or script.
+        /// AnalyzeDSCResource: Analyzes the ast to check that Write-Verbose is called for DSC Resources
         /// <param name="ast">The script's ast</param>
         /// <param name="fileName">The script's file name</param>
         /// </summary>
-        public IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string fileName)
+        public IEnumerable<DiagnosticRecord> AnalyzeDSCResource(Ast ast, string fileName)
         {
-            if (ast == null) throw new ArgumentNullException(Strings.NullAstErrorMessage);
-            
-            ClearList();
-            this.AddNames(new List<string>() { "Configuration", "Workflow" });
-            DiagnosticRecords.Clear();
-            
-            this.fileName = fileName;
-            //We only check that advanced functions should have Write-Verbose
-            ast.Visit(this);
-
-            return DiagnosticRecords;
-        }
-
-        /// <summary>
-        /// Visit function and checks that it has write verbose
-        /// </summary>
-        /// <param name="funcAst"></param>
-        /// <returns></returns>
-        public override AstVisitAction VisitFunctionDefinition(FunctionDefinitionAst funcAst)
-        {
-            if (funcAst == null)
+            if (ast == null)
             {
-                return AstVisitAction.SkipChildren;
+                throw new ArgumentNullException(Strings.NullAstErrorMessage);
             }
 
-            //Write-Verbose is not required for non-advanced functions
-            if (funcAst.Body == null || funcAst.Body.ParamBlock == null
-           || funcAst.Body.ParamBlock.Attributes == null ||
-           funcAst.Body.ParamBlock.Parameters == null ||
-                !funcAst.Body.ParamBlock.Attributes.Any(attr => attr.TypeName.GetReflectionType() == typeof(CmdletBindingAttribute)))
-            {
-                return AstVisitAction.Continue;
-            }
+            List<string> expectedTargetResourceFunctionNames = new List<string>(new string[] { "Set-TargetResource", "Test-TargetResource", "Get-TargetResource" });            
```

Comment:
Quoc. I did not understand your comment.

Since this rule is enabled only for cmdlet based DSC resources, we do not need to validate for advanced functions...